config/routes.rb

```diff
@@ -160,13 +160,16 @@
 
       post '/convert', to: 'converts#create'
 
-      get '/races',       to: 'races/races#index'
-      get '/bingos',      to: 'races/bingos#index'
-      get '/randomizers', to: 'races/randomizers#index'
-
-      get '/races/:race',       to: 'races/races#show'
-      get '/bingos/:race',      to: 'races/bingos#show'
-      get '/randomizers/:race', to: 'races/randomizers#show'
+      Raceable::RACE_TYPES.map { |raceable| raceable.type.to_s.pluralize }.each do |type|
+        get   "/#{type}",       to: "races/#{type}#index"
+        post  "/#{type}",       to: "races/#{type}#create"
+        get   "/#{type}/:race", to: "races/#{type}#show"
+        patch "/#{type}/:race", to: "races/#{type}#update"
+
+        post   "/#{type}/:race/entrants",          to: "races/entrants/#{type}#create"
+        patch  "/#{type}/:race/entrants/:entrant", to: "races/entrants/#{type}#update"
+        delete "/#{type}/:race/entrants/:entrant", to: "races/entrants/#{type}#destroy"
```

Comment:
Thoughts on doing PATCH/DELETE `/#{type}/:race/entrant` instead, and inferring the entrant based on auth? Can't see us ever needing to have one account manage multiple entrants in the same race, and this would make it simpler for consumers (including our frontend), and less prone to auth bugs on the backend.

Reply:
I actually like it more.  I thought about it but it seemed less REST-ful but if you don't think that's a problem I'm more than willing to head that direction.  Inferring the entrant also means clients don't have to track their ID's, just manage the state that they're currently in.